middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -3813,10 +3813,53 @@ let bind_static_consts_and_code acc body =
     (acc, body) components
 
 let wrap_final_module_block acc env ~program ~prog_return_cont
-    ~module_block_size_in_words ~return_cont ~module_symbol =
+    ~(module_repr : Lambda.module_representation) ~return_cont ~module_symbol =
   let module_block_var = Variable.create "module_block" K.value in
   let module_block_var_duid = Flambda_debug_uid.none in
   let module_block_tag = Tag.Scannable.zero in
+  let block_shape, field_count, block_access, kind_of_field =
```

Comment:
The body of this let feels natural to pull out into a helper function called something like `shape_of_final_module_block` (or something like that).  This will help keep `wrap_final_module_block` readable.